Entities/Industry/Fireplace/Fireplace.as

```diff
@@ -53,6 +54,10 @@ void onCollision(CBlob@ this, CBlob@ blob, bool solid)
 			food.setVelocity(blob.getVelocity().opMul(0.5f));
 		}
 	}
+	else if (blob !is null && blob.hasTag("fire source")) //fire arrow works
```

Comment:
Move the blob null check outside of this if statement so it isn't being repeated by both if statements.